clients/src/main/java/org/apache/kafka/common/network/SSLTransportLayer.java

```diff
@@ -222,11 +222,13 @@ public void handshake() throws IOException {
                               channelId, appReadBuffer.position(), netReadBuffer.position(), netWriteBuffer.position());
                     handshakeResult = handshakeWrap(write);
                     if (handshakeResult.getStatus() == Status.BUFFER_OVERFLOW) {
-                        int currentPacketBufferSize = packetBufferSize();
-                        netWriteBuffer = Utils.ensureCapacity(netWriteBuffer, currentPacketBufferSize);
-                        if (netWriteBuffer.position() >= currentPacketBufferSize) {
+                        int currentNetWriteBufferSize = netWriteBufferSize();
+                        netWriteBuffer.position(netWriteBuffer.limit());
```

Comment:
That makes sense. What about the scenario where `flush` in `handshakeWrap` writes some data into the socket buffer, could we have something like the following?

1. wrap() completes with some data in the buffer (eg. position=15)
2. handshakeWrap() flips the buffer (limit=15, position=0)
3. flush writes some data to the buffer (limit=15, position=5)
4. Set position to limit (limit=15, position=15)
5. Utils.ensureCapacity() flips the buffer again (limit=15, position=0)

In this case, it seems like we would lose the position of the original buffer or am I missing something?